dwave/optimization/include/dwave-optimization/nodes/lp.hpp

```diff
@@ -56,7 +56,32 @@ class FeasibleNode : public ScalarOutputMixin<ArrayNode> {
     void revert(State& state) const override;
 
  private:
-    const LPNode* lp_ptr_;
+    const LPNodeBase* lp_ptr_;
+};
+
+class LPNodeBase : public Node {
+ public:
+    /// The default lower bound for variables
+    static const double default_lower_bound();
+
+    /// The default lower bound for variables
+    static const double default_upper_bound();
+
+    /// Return whether the state is feasible for the LP model.
+    virtual bool feasible(const State& state) const = 0;
+
+    /// Any upper bound equal to or greater (or lower bound less to or equal)
+    /// to this will be treated as unbounded.
+    static const double infinity();
+
+    /// Return the objective value. Undefined when the LP is not feasible.
+    virtual double objective_value(const State& state) const = 0;
+
+    virtual std::span<const double> solution(const State& state) const = 0;
+
+    virtual std::span<const ssize_t> variables_shape() const = 0;
+
+    virtual std::pair<double, double> _minmax() const = 0;
```

Comment:
Why the leading underscore?

Reply:
I wanted to differentiate from the standard `minmax` but I'm realizing `variables_minmax()` is more appropriate.